src/index.d.ts

```diff
@@ -0,0 +1,395 @@
+         * @param  {boolean} shared
+         * @returns void
+         */
+        setLocationShared(shared: boolean): void;
+
+        /**
+         * Prompts the user for location permissions to allow geotagging from the OneSignal dashboard.
+         * @returns void
+         */
+        promptLocation(): void;
+
+        /**
+         * This method returns a "snapshot" of the device state for when it was called.
+         * @returns Promise<DeviceState>
+         */
+        getDeviceState(): Promise<DeviceState>;
+
+        /**
+         * Did the user provide privacy consent for GDPR purposes.
+         * @returns Promise<boolean>
+         */
+        userProvidedPrivacyConsent(): Promise<boolean>;
+
+        /**
+         * Tag a user based on an app event of your choosing so they can be targeted later via segments.
+         * @param  {string} key
+         * @param  {string} value
+         * @returns void
+         */
+        sendTag(key: string, value: string): void;
+
+        /**
+         * Tag a user wiht multiple tags based on an app event of your choosing so they can be targeted later via segments.
+         * @param  {object} tags
+         * @returns void
+         */
+        sendTags(tags: object): void;
+
+        /**
+         * Retrieve a list of tags that have been set on the user from the OneSignal server.
+         * @param  {(tags:object)=>void} handler
+         * @returns void
+         */
+        getTags(handler: (tags: object) => void): void;
+
+        /**
+         * Deletes a single tag that was previously set on a user.
+         * @param  {string} key
+         * @returns void
+         */
+        deleteTag(key: string): void;
+
+        /**
+         * Deletes multiple tags that were previously set on a user.
+         * @param  {string[]} keys
+         */
+        deleteTags(keys: string[]);
+
+        /**
+         * Allows you to set the user's email address with the OneSignal SDK.
+         * @param  {string} email
+         * @param  {string} authCode
+         * @param  {Function} handler
+         * @returns void
+         */
+        setEmail(email: string, authCode: string, handler: Function): void;
+
+        /**
+         * If your app implements logout functionality, you can call logoutEmail to dissociate the email from the device.
+         * @param  {Function} handler
+         */
+        logoutEmail(handler: Function);
+
+        /**
+         * Send a notification
+         * @param  {string} notificationObjectString - JSON string payload (see REST API reference)
+         * @param  {(success:object)=>void} onSuccess
+         * @param  {(failure:object)=>void} onFailure
+         * @returns void
+         */
+        postNotification(notificationObjectString: string, onSuccess: (success: object) => void, onFailure: (failure: object) => void): void;
+
+        /**
+         * iOS provides a standard way to clear notifications by clearing badge count.
```

Comment:
I believe we changed this wording in the OneSignal public docs. It should match that

Reply:
[From docs](https://documentation.onesignal.com/docs/ios-native-sdk#clearonesignalnotifications):

> iOS provides a standard way to clear notifications by clearing badge count, thus there is no specific OneSignal API call for clearing notifications.

Reply:
I was thinking this one as it does indicate it is only for Android.
> Android only. iOS provides a standard way to clear notifications by clearing badge count. There is no specific OneSignal API call for clearing notifications.
https://documentation.onesignal.com/docs/sdk-reference#clearonesignalnotifications-method

I am ok with either description for now in this PR but thinking we just need to add "only Android" for sure.

Reply:
Done.